src/main/java/org/assertj/core/internal/Comparables.java

```diff
@@ -255,12 +256,16 @@ private boolean isLessThan(Object actual, Object other) {
    * @throws AssertionError if the actual value is not between start and end.
    * @throws NullPointerException if start value is {@code null}.
    * @throws NullPointerException if end value is {@code null}.
+   * @throws IllegalArgumentException if end value is less than start value.
    */
   public <T extends Comparable<? super T>> void assertIsBetween(AssertionInfo info, T actual, T start, T end,
       boolean inclusiveStart, boolean inclusiveEnd) {
     assertNotNull(info, actual);
     checkNotNull(start, "The start range to compare actual with should not be null");
     checkNotNull(end, "The end range to compare actual with should not be null");
+    checkArgument(comparisonStrategy.isLessThanOrEqualTo(start, end),
```

Comment:
@joel-costigliola thx, currently we end up in a `AssertionError` in such a case. If this change is ok I can change the implementation to also throw an `IllegalArgumentException` in this case.